src/main/java/eu/dissco/backend/repository/ElasticSearchRepository.java

```diff
@@ -105,27 +112,23 @@ private AnnotationResponse mapToAnnotationResponse(ObjectNode json) {
         null);
   }
 
-  private DigitalSpecimen mapToDigitalSpecimen(ObjectNode json) {
-    var digitalSpecimen = json.get("digitalSpecimen");
-    var attributes = digitalSpecimen.get("ods:attributes");
-    addUrlToAttributes(attributes);
-    var createdOn = parseDate(json.get(FIELD_CREATED));
-    return new DigitalSpecimen(
-        HANDLE_STRING + json.get("id").asText(),
-        json.get("midsLevel").asInt(),
-        json.get("version").asInt(),
-        createdOn,
-        getText(digitalSpecimen, "ods:type"),
-        getText(digitalSpecimen, "ods:physicalSpecimenId"),
-        getText(attributes, "ods:physicalSpecimenIdType"),
-        getText(attributes, "ods:specimenName"),
-        getText(attributes, "ods:organisationId"),
-        getText(attributes, "ods:datasetId"),
-        getText(attributes, "ods:physicalSpecimenCollection"),
-        getText(attributes, "ods:sourceSystemId"),
-        attributes,
-        digitalSpecimen.get("ods:originalAttributes"),
-        getText(attributes, "dwca:id"));
+  private DigitalSpecimenWrapper mapToDigitalSpecimen(ObjectNode json)
+      throws DiSSCoElasticMappingException {
```

Comment:
Because this is a runtime exception, you don't need to declare it in the header

Reply:
:+1: